src/jit/codegenarmarch.cpp

```diff
@@ -3500,11 +3486,20 @@ void CodeGen::genCodeForStoreBlk(GenTreeBlk* blkOp)
 void CodeGen::genScaledAdd(emitAttr attr, regNumber targetReg, regNumber baseReg, regNumber indexReg, int scale)
 {
     emitter* emit = getEmitter();
+    if (scale == 0)
+    {
+        // target = base + index
+        getEmitter()->emitIns_R_R_R(INS_add, attr, targetReg, baseReg, indexReg);
+    }
+    else
+    {
```

Comment:
Why is it worth special casing scale == 0?

Reply:
The emitter currently doesn't allow 0 to be used with INS_OPTS_LSL:

```
        else if (insOptsAluShift(opt))
        {
            // imm should be non-zero and in [1..63]
            assert(isValidImmShift(imm, size) && (imm != 0));
            fmt = IF_DR_3B;
        }
```